provider/gandi/gandi.go

```diff
@@ -163,14 +209,29 @@ func (p *GandiProvider) submitChanges(ctx context.Context, changes []*GandiChang
 
 	for _, changes := range zoneChanges {
 		for _, change := range changes {
+			if change.Record.RrsetType == endpoint.RecordTypeCNAME && !strings.HasSuffix(change.Record.RrsetValues[0], ".") {
+				change.Record.RrsetValues[0] += "."
+			}
+
 			// Prepare record name
 			recordName := strings.TrimSuffix(change.Record.RrsetName, "."+change.ZoneName)
 			if recordName == change.ZoneName {
```

Comment:
If the zone is `example.com` and the `change.Record.RrsetName` is `example.com.example.com`, then this code will incorrectly replace `recordName` with `@`.

Reply:
To reduce noise on this PR, I have fixed this in #3893